arm/postgresql/models.go

```diff
@@ -0,0 +1,291 @@
+type Operation struct {
+	Name       *string                             `json:"name,omitempty"`
+	Display    *OperationDisplay                   `json:"display,omitempty"`
+	Origin     OperationOrigin                     `json:"origin,omitempty"`
+	Properties *map[string]*map[string]interface{} `json:"properties,omitempty"`
+}
+
+// OperationDisplay is display metadata associated with the operation.
+type OperationDisplay struct {
+	Provider    *string `json:"provider,omitempty"`
+	Resource    *string `json:"resource,omitempty"`
+	Operation   *string `json:"operation,omitempty"`
+	Description *string `json:"description,omitempty"`
+}
+
+// OperationListResult is a list of resource provider operations.
+type OperationListResult struct {
+	autorest.Response `json:"-"`
+	Value             *[]Operation `json:"value,omitempty"`
+}
+
+// ProxyResource is resource properties.
+type ProxyResource struct {
+	ID   *string `json:"id,omitempty"`
+	Name *string `json:"name,omitempty"`
+	Type *string `json:"type,omitempty"`
+}
+
+// Server is represents a server.
+type Server struct {
+	autorest.Response `json:"-"`
+	ID                *string             `json:"id,omitempty"`
+	Name              *string             `json:"name,omitempty"`
+	Type              *string             `json:"type,omitempty"`
+	Location          *string             `json:"location,omitempty"`
+	Tags              *map[string]*string `json:"tags,omitempty"`
+	Sku               *Sku                `json:"sku,omitempty"`
+	*ServerProperties `json:"properties,omitempty"`
+}
+
+// ServerForCreate is represents a server to be created.
+type ServerForCreate struct {
+	Sku        *Sku                       `json:"sku,omitempty"`
+	Properties *ServerPropertiesForCreate `json:"properties,omitempty"`
+	Location   *string                    `json:"location,omitempty"`
+	Tags       *map[string]*string        `json:"tags,omitempty"`
+}
+
+// ServerListResult is a list of servers.
+type ServerListResult struct {
+	autorest.Response `json:"-"`
+	Value             *[]Server `json:"value,omitempty"`
+}
+
+// ServerProperties is the properties of a server.
+type ServerProperties struct {
+	AdministratorLogin       *string            `json:"administratorLogin,omitempty"`
+	StorageMB                *int64             `json:"storageMB,omitempty"`
+	Version                  ServerVersion      `json:"version,omitempty"`
+	SslEnforcement           SslEnforcementEnum `json:"sslEnforcement,omitempty"`
+	UserVisibleState         ServerState        `json:"userVisibleState,omitempty"`
+	FullyQualifiedDomainName *string            `json:"fullyQualifiedDomainName,omitempty"`
+}
+
+// ServerPropertiesForCreate is the properties used to create a new server.
+type ServerPropertiesForCreate struct {
+	StorageMB      *int64             `json:"storageMB,omitempty"`
+	Version        ServerVersion      `json:"version,omitempty"`
+	SslEnforcement SslEnforcementEnum `json:"sslEnforcement,omitempty"`
```

Comment:
just a heads up that I've opened https://github.com/Azure/azure-rest-api-specs/pull/1391 to add the fields `administratorLogin` and `administratorLoginPassword` given these fields are required (I can't tell if it's incorrect Swagger or if the Go Generator doesn't support the Swagger extension being used, due to the class below)